datanode/entities/marketdata.go

```diff
@@ -411,10 +420,10 @@ func priceMonitoringBoundsToProto(bounds []*PriceMonitoringBound) []*types.Price
 
 	for _, bound := range bounds {
 		protoBound := types.PriceMonitoringBounds{
-			MinValidPrice:  fmt.Sprintf("%d", bound.MinValidPrice),
-			MaxValidPrice:  fmt.Sprintf("%d", bound.MaxValidPrice),
+			MinValidPrice:  bound.MinValidPrice.String(),
+			MaxValidPrice:  bound.MaxValidPrice.String(),
 			Trigger:        priceMonitoringTriggerToProto(bound.Trigger),
-			ReferencePrice: fmt.Sprintf("%d", bound.ReferencePrice),
+			ReferencePrice: bound.ReferencePrice.String(),
```

Comment:
If these returned an error earlier you could be call .String() on a nil pointer.